cmake/OpenCVUtils.cmake

```diff
@@ -309,7 +309,11 @@ function(ocv_include_directories)
            dir MATCHES "/usr/include$")
       # workaround for GCC 6.x bug
     else()
-      include_directories(AFTER SYSTEM "${dir}")
+      if(${CMAKE_SYSTEM_NAME} MATCHES QNX)
+        include_directories(AFTER "${dir}")
+      else()
+        include_directories(AFTER SYSTEM "${dir}")
```

Comment:
What is the problem? message?

If there is no SYSTEM headers support then it should be disabled on CMake level.

Reply:
```console
cd /home/chachoi/chachoi-world/qnx-ports/opencv/nto-aarch64-le/build/modules/imgcodecs && /home/chachoi/qnx800/host/linux/x86_64/usr/bin/qcc -lang-c++ -DCVAPI_EXPORTS -DHAVE_IMGCODEC_HDR -DHAVE_IMGCODEC_PFM -DHAVE_IMGCODEC_PXM -DHAVE_IMGCODEC_SUNRASTER -DHAVE_WEBP -DOPJ_STATIC -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/chachoi/chachoi-world/opencv/3rdparty/openjpeg/openjp2 -I/home/chachoi/chachoi-world/qnx-ports/opencv/nto-aarch64-le/build/3rdparty/openjpeg/openjp2 -I/home/chachoi/chachoi-world/opencv/3rdparty/libwebp/src -I/home/chachoi/chachoi-world/opencv/modules/imgcodecs/include -I/home/chachoi/chachoi-world/qnx-ports/opencv/nto-aarch64-le/build/modules/imgcodecs -I/home/chachoi/chachoi-world/opencv/modules/core/include -I/home/chachoi/chachoi-world/opencv/modules/imgproc/include -Wp,-isystem,/home/chachoi/chachoi-world/qnx-ports/opencv/nto-aarch64-le/build -Wp,-isystem,/home/chachoi/qnx800/target/qnx/usr/include -Vgcc_ntoaarch64le   -I. -I/home/chachoi/chachoi-world/qnx-ports/opencv/nto-aarch64-le -I/home/chachoi/chachoi-world/qnx-ports/opencv -I/home/chachoi/qnx800/target/qnx/usr/include   -I/home/chachoi/qnx800/target/qnx/aarch64le//usr/local/include -D_QNX_SOURCE   -fsigned-char    -O3 -DNDEBUG -O3 -DNDEBUG  -fPIC -std=c++17 -o CMakeFiles/opencv_imgcodecs.dir/src/loadsave.cpp.o -c /home/chachoi/chachoi-world/opencv/modules/imgcodecs/src/loadsave.cpp
cc: warning - lang-c++ is deprecated
In file included from /home/chachoi/qnx800/target/qnx/usr/include/c++/v1/__type_traits/remove_reference.h:13,
                 from /home/chachoi/qnx800/target/qnx/usr/include/c++/v1/__type_traits/apply_cv.h:15,
                 from /home/chachoi/qnx800/target/qnx/usr/include/c++/v1/__functional/invoke.h:15,
                 from /home/chachoi/qnx800/target/qnx/usr/include/c++/v1/type_traits:421,
                 from /home/chachoi/qnx800/target/qnx/usr/include/c++/v1/limits:827,
                 from /home/chachoi/chachoi-world/opencv/modules/core/include/opencv2/core/cvdef.h:185,
                 from /home/chachoi/chachoi-world/opencv/modules/core/include/opencv2/core.hpp:52,
                 from /home/chachoi/chachoi-world/opencv/modules/imgcodecs/include/opencv2/imgcodecs.hpp:46,
                 from /home/chachoi/chachoi-world/opencv/modules/imgcodecs/src/precomp.hpp:45,
                 from /home/chachoi/chachoi-world/opencv/modules/imgcodecs/src/loadsave.cpp:46:
/home/chachoi/qnx800/target/qnx/usr/include/c++/v1/cstddef:46:5: error: #error <cstddef> tried including <stddef.h> but didn't find libc++'s <stddef.h> header. This usually means that your header search paths are not configured properly. The header search paths should contain the C++ Standard Library headers before any C Standard Library, and you are probably using compiler flags that make that not be the case.
   46 | #   error <cstddef> tried including <stddef.h> but didn't find libc++'s <stddef.h> header. \
```

QNX headers at `-isystem,/home/chachoi/qnx800/target/qnx/usr/include` being treated as system headers causes the compile issue above. I tried setting `CMAKE_NO_SYSTEM_FROM_IMPORTED` to [`ON`](https://github.com/chachoi-world/qnx-ports/blob/4a2bca8505f9c505cb6dc356d01f9a0e87695beb/opencv/common.mk#L75), but it did not work.